### DISCUSSION TITLE
EfficientAd is slower than other models in anomalib

### DISCUSSION THREAD

**Discussion by haimat** [open-edge-platform/anomalib#2147] [Feature Requests]
I had the impression that the EfficientAd model would be among the fastest in anomalib in terms of prediction times. To verify that I have trained three models, Padim, Fastflow, and EfficientAd, all with the same training data and an image dimension of 512x512 pixels. Then I have written a small script that loads these models, warms up the GPU, and then runs prediction on 100 images. I measure only the model forward time, no image loading or any pre- or post-processing.

With the models exported to ONNX I get these results (avg. model forwards times on 100 images):
- EfficientAd: 0.0116 sec.
- Fastflow: 0.0053 sec.
- Padim: 0.0036 sec.

So in other words: The EfficientAd model is the slowest from these three, and Padim the fastest - I thought it would be the other way round. Am I missing something, or is this a bug in anomalib?

**Comment by haimat** [comment]
Hello, no news on this?

**Reply by Iheb-prepa** [reply]
To my knowledge, there's no official code for EfficientAd, however you can check this [unofficial implementation](https://github.com/nelson1425/EfficientAD) which claims to match the paper's results.

**Comment by abc-125** [comment]
Hello, I have a couple of suggestions. You can try 256x256 pixels size to match papers. Also, you can try to run it without ONNX, optimization might be working differently for different models.

**Comment by Issa-Bqain** [comment]
@haimat Can you please share the source code you are using for this comparison and I can see if these results are reproducible on my end? thanks

**Reply by haimat** [reply]
> @haimat Can you please share the source code you are using for this comparison and I can see if these results are reproducible on my end? thanks

Unfortunately not, I don't have the code any more, it was an ad-hoc test.
But quite easy to reproduce, as described in my original posting.
